boards/arm/nucleo_g431rb/doc/index.rst

```diff
@@ -0,0 +1,236 @@
+
+The Zephyr nucleo_g431rb board configuration supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| I2C       | on-chip    | i2c                                 |
++-----------+------------+-------------------------------------+
+| PWM       | on-chip    | pwm                                 |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on this Zephyr port.
+
+The default configuration can be found in the defconfig file:
+``boards/arm/nucleo_g431rb/nucleo_g431rb_defconfig``
+
+
+Connections and IOs
+===================
+
+Nucleo G431RB Board has 6 GPIO controllers. These controllers are responsible for pin muxing,
+input/output, pull-up, etc.
+
+For mode details please refer to `STM32G4 Nucleo-64 board User Manual`_.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- UART_1_TX : PC4
+- UART_1_RX : PC5
+- UART_2_TX : PA2
+- UART_2_RX : PA15
+- I2C_1_SCL : PB8
+- I2C_1_SDA : PB9
+- SPI_1_NSS : PB6
+- SPI_1_SCK : PB3
+- SPI_1_MISO : PA6
+- SPI_1_MOSI : PA7
+- SPI_2_NSS : PB12
+- SPI_2_SCK : PB13
+- SPI_2_MISO : PB14
+- SPI_2_MOSI : PB15
+- SPI_3_NSS : PA15
+- SPI_3_SCK : PC10
+- SPI_3_MISO : PC11
+- SPI_3_MOSI : PC12
+- PWM_3_CH1 : PB4
+- USER_PB : PC13
+- LD2 : PA5
+- USB DM : PA11
+- USB DP : PA12
+
+System Clock
+------------
+
+Nucleo G431RB System Clock could be driven by internal or external oscillator,
+as well as main PLL clock. By default System clock is driven by PLL clock at 150MHz,
+driven by 16MHz high speed internal oscillator. The clock can be boosted to 170MHz if boost mode
+is selected.
+
+Serial Port
+-----------
+
+Nucleo G431RB board has 3 U(S)ARTs. The Zephyr console output is assigned to UART2.
+Default settings are 115200 8N1.
+
+
+Programming and Debugging
+*************************
+
+Applications for the ``nucleo_g431rb`` board configuration can be built and
+flashed in the usual way (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Flashing
+========
+
+Nucleo G431RB board includes an ST-LINK/V3E embedded debug tool
+interface.  This interface is supported by the openocd version
+included in the Zephyr SDK since v0.10.4.
```

Comment:
Is it ?
Current Zephyr SDK is 0.10.3. And for now, I don't see any g4 support in openocd upstream (unfortunately)

Reply:
I submitted this but I don't think it's going to go anywhere https://github.com/zephyrproject-rtos/openocd/pull/11